protocol.go

```diff
@@ -0,0 +1,9 @@
+package mssql
+
+type Protocol string
+
+const (
+	TCP           Protocol = "tcp:"
```

Comment:
Constants in to should begin with a prefix and not be in all caps:
protoTCP
protoNamedPipe
protoSharedMemory

The Protocol type probably shouldn't be exported.